src/main/java/Containers/Portfolio.java

```diff
@@ -23,67 +23,116 @@ public static Portfolio getInstance() {
         return Portfolio.commonPortfolio;
     }
 
-    // Storage: The UUID of asset and the reference of asset
-    private final HashMap<String, Asset> storage;
+
+
+    // assetList: a list of asset in the portfolio
+    private List<Asset> assetList;
+    // transactionList: a list of transaction in the portfolio
+    private List<Transaction> transactionList;
+    // votingHistory: a history of votes in the portfolio
+    private List<Vote> votingHistory;
+    private double profitability;
 
     private Portfolio() {
-        this.storage = new LinkedHashMap<String, Asset>();
+        this.assetList = new ArrayList<>();
+        this.transactionList = new ArrayList<>();
+        this.votingHistory = new ArrayList<>();
+        this.profitability = 0.0;
+    }
+    // Getter and Setter for profitability
+    public double getProfitability() {
+        return profitability;
+    }
+
+    public void setProfitability(double profitability) {
+        this.profitability = profitability;
+    }
+
+    // Getter and Setter for assets, transactions, and votes
+
+    public List<Asset> getAssetList() {
+        return assetList;
+    }
+
+    public List<Transaction> getTransactionList() {
+        return transactionList;
+    }
+
+    public List<Vote> getVotingHistory() {
+        return votingHistory;
     }
 
     // Add an asset to the system.
     // This method only takes a snapshot of its parameter, the parameter object can be safely modified afterwards.
     public void add(Asset asset0) {
-        Asset asset = this.storage.get(asset0.getType());
-        if(asset == null) {
-            asset = new Asset(0, asset0.getPrice(), asset0.getType(), asset0.getSymbol());
-            this.storage.put(asset.getType(), asset);
-        }
-        asset.setVolume(asset.getVolume() + asset0.getVolume());
+        this.assetList.add(asset0);
+    }
+
+    // Add an asset to the system.
+    public void add(Transaction transaction0) {
+        this.transactionList.add(transaction0);
+    }
+
+    // Add a vote to the system.
```

Comment:
"Add a vote to the system" is a bit inaccurate. I would write "Record a vote in the portfolio's vote history". Also this method should probably have a javadoc.

Reply:
Also, vote0 as a parameter name is a bit awkward. I think just "vote" would suffice.